monai/auto3dseg/analyzer.py

```diff
@@ -229,29 +229,31 @@ def __call__(self, data):
         """
         d = dict(data)
         start = time.time()
-        ndas = data[self.image_key]
-        ndas = [ndas[i] for i in range(ndas.shape[0])]
-        if "nda_croppeds" not in d:
-            nda_croppeds = [get_foreground_image(nda) for nda in ndas]
 
-        # perform calculation
-        report = deepcopy(self.get_report_format())
+        with torch.no_grad():
```

Comment:
I'm not sure if it's needed to add `no_grad()`  for each analyzer, it looks like lots of lines changed. how about we add it here in the summarizer? if it doesn't work, we make further changes?
https://github.com/Project-MONAI/MONAI/blob/1f9e5ce34691af8be20a84a673ecf5c7535e6b66/monai/auto3dseg/seg_summarizer.py#L181

Reply:
Thanks @wyli for the suggestion. Lots of line changes are due to the indentation. To reduce the number of line changes, I would replace `with torch.no_grad()` with

```
torch.set_grad_enabled(mode)
```

Adding it in the summarizer also looked like a good idea to me first, but later I found a problem: the summarizer inherits `Compose`. The computation happens in the `__call__` functions of `Compose` and adding `torch.no_grad()` there seemed a bad idea.